source/_posts/2015-10-09-the-slack-users-code.markdown

```diff
@@ -0,0 +1,258 @@
+  administrator, consider using [Slack's custom profiles](http://slackhq.com/post/132116511325/customprofiles)
+  to make a special space for this information.
+
+
+## Channels
+
+Channels, especially, are important to name in a consistent and predictable way.
+You want channels to be discoverable and obvious so that if someone is trying to
+find the right channel to ask a question or start a discussion in, they have a
+good chance of doing so.
+
+
+### General Guidelines
+
+#### Rename The #general Channel
+
+This one's just for administrators, really. Consider renaming #general to
+something else. It's a special channel that everyone _must_ be in. No one can
+`/leave` it. It's important for people to understand that talking at all in
+ #general can be bothersome to a lot of folks, let alone using @here or @channel.
+Renaming it to #everyone or #all-hands or something might make that more clear.
+In a large enough organization, locking down this channel so that only
+administrators can talk in it (and rarely at that) is a very, very good idea.
+
+Corollary: "General" isn't a very descriptive word. You don't need to use it in
+channel names. #product-general isn't really any more descriptive than #product.
+
+#### Don't Create Private Channels
+
+You might think no one else cares. If they don't care, they don't have to join
+your channel. Private channels only block communication and discoverability.
+Your default assumption should be that all channels should be public. If you
+have something truly secret to discuss—something that even other employees of
+your company shouldn't see—consider that you're sharing anything you type into
+Slack with another company. If it's truly secret, consider sending PGP signed
+emails or only talking face-to-face. Much of what you do probably isn't actually
+that secret. Flowdock has [a great post about the dangers of private channels](http://blog.flowdock.com/2014/04/30/beware-of-private-conversations/).
+Also see [Zach Holman's fine writing about opt-in transparency](http://zachholman.com/posts/opt-in-transparency/).
+
+
+#### Don't Create Extraneous Channels
+
+You might think you're bothering people in another channel. For some kinds of
+channels (see below) it might be appropriate to create, but in general, you
+should wait until some sub-topic is routinely taking over a more general channel
+before you create a new one You don't want to get into a situation where there
+are more channels than people in your Slack team. Specifically, a team should
+not create a new channel for every individual problem they want to discuss. Keep
+things in as general a channel as folks can stand.
+
+
+#### Use Dashes in Channel Names
+
+Just for consistency. Do #austin-lunch-games not #austin_lunch_games or any
+other ridiculousness.
+
+
+#### Know What Kind of Channel You're Creating
+
+There are two broad categories of channel: Work-Related and Off-Topic.
```

Comment:
Personally, I’d use italics to denote the two hyphenated terms rather than artificial capitalization, but this is purely a matter of taste.